src/views/collection-view/components/collection-meta-progress/helpers/parse-collection-progress.ts

```diff
@@ -0,0 +1,155 @@
+ * and a count of all tutorials in that collection,
+ * Return booleans representing collection progress state.
+ */
+function parseCollectionProgress(
+	progressData: ApiCollectionTutorialProgress[],
+	tutorials: TutorialLite[],
+	collection: Pick<Collection, 'id' | 'slug'>
+) {
+	/**
+	 * The basics
+	 */
+	const tutorialCount = tutorials.length
+	const completedTutorialCount = countCompletedRecords(
+		progressData || [],
+		collection.id
+	)
+	const isCompleted = completedTutorialCount == tutorialCount
+	const isInProgress = completedTutorialCount > 0 && !isCompleted
+	/**
+	 * Tutorial CTA
+	 */
+	const tutorialCta = getNextTutorialCta({
+		progressData,
+		tutorials,
+		collectionSlug: collection.slug,
+		isCompleted,
+		isInProgress,
+		completedTutorialCount,
+		tutorialCount,
+	})
+	/**
+	 * Status label
+	 */
+	let statusLabel: string
+	if (isCompleted) {
+		statusLabel = 'Complete'
+	} else if (isInProgress) {
+		statusLabel = `${completedTutorialCount}/${tutorialCount}`
+	} else {
+		statusLabel = `${tutorialCount} tutorial${tutorialCount == 1 ? '' : 's'}`
+	}
+	/**
+	 * Return it all
+	 */
+	return {
+		completedTutorialCount,
+		isCompleted,
+		isInProgress,
+		statusLabel,
+		tutorialCount,
+		tutorialCta,
+	}
+}
+
+/**
+ * Given an array of progress records, and a collection.id,
+ * Return the count of records that are completed in that collection.
+ */
+function countCompletedRecords(
+	progressData: ApiCollectionTutorialProgress[],
+	collectionId: Collection['id']
+): number {
+	return progressData.filter((record: ApiCollectionTutorialProgress) => {
+		return (
+			record.collection_id == collectionId && record.complete_percent == '100'
+		)
+	}).length
+}
+
+/**
+ * Given progress and tutorial data for particular collection,
+ * Return a CTA object that links to the "next tutorial" in the collection.
+ *
+ * For in-progress collections, the "next tutorial" is the first tutorial
+ * in the collection that does not have "complete" as its progress status.
+ *
+ * For completed collections, and for all unauthenticated cases,
+ * the "next tutorial" is the first tutorial in the collection.
+ */
+function getNextTutorialCta({
+	progressData,
+	tutorials,
+	isCompleted,
+	isInProgress,
+	collectionSlug,
+	completedTutorialCount,
+	tutorialCount,
+}: {
+	progressData: ApiCollectionTutorialProgress[]
+	tutorials: TutorialLite[]
+	isCompleted: boolean
+	isInProgress: boolean
+	collectionSlug: Collection['slug']
+	completedTutorialCount: number
+	tutorialCount: number
+}) {
+	/**
+	 * If we're not authenticated or have no progress,
+	 * we'll use the first tutorial.
+	 *
+	 * If we do have progress data, we return the first incomplete tutorial,
+	 * or if all tutorials are complete, then the first tutorial.
+	 */
+	let targetTutorial
+	if (isCompleted || !progressData) {
+		targetTutorial = tutorials[0]
+	} else {
+		const firstIncompleteTutorial = tutorials.find((tutorial: TutorialLite) => {
```

Comment:
For some reason this isn't behaving as it should while I test it. The logic looks right but when I test, it always links to the first tutorial even when there are other tutorials in progress and the first tutorial hasn't been visited.

Reply:
Ah okay yeah I interpreted this wrong - I'd interpreted it as "go to the first tutorial that's not completed" rather than strictly "the first tutorial that's in progress". 

I'll adjust the logic, and rename this to `firstInProgressTutorial` (and then fall back to the first tutorial if there are no `in_progress` tutorials 👍)

Reply:
✅ Updated - should now work as expected, I think. Also updated the logic for `isInProgress`, so that we see the progress bar even if `0/x` tutorials are complete. Thanks for the vid 💯 🙇